workflows/environment_upsert.go

```diff
@@ -246,66 +238,51 @@ func (workflow *environmentWorkflow) environmentUpserter(namespace string, ecsSt
 
 		var templateName string
 		var imagePattern string
-		if environment.Provider == common.EnvProviderEcs {
-			templateName = "env-ecs.yml"
-			imagePattern = ecsImagePattern
-			stackParams["LaunchType"] = "EC2"
-		} else if environment.Provider == common.EnvProviderEcsFargate {
-			templateName = "env-ecs.yml"
-			imagePattern = ecsImagePattern
-			stackParams["LaunchType"] = "FARGATE"
-		} else if environment.Provider == common.EnvProviderEc2 {
-			templateName = "env-ec2.yml"
-			imagePattern = ec2ImagePattern
-		}
+		envMapping := map[common.EnvProvider]map[string]string{
+			common.EnvProviderEcs: map[string]string{
+				"templateName": "env-ecs.yml",
+				"imagePattern": ecsImagePattern,
+				"launchType":   "EC2"},
+			common.EnvProviderEcsFargate: map[string]string{
+				"templateName": "env-ecs.yml",
+				"imagePattern": ecsImagePattern,
+				"launchType":   "FARGATE"},
+			common.EnvProviderEc2: map[string]string{
+				"templateName": "env-ec2.yml",
+				"imagePattern": ec2ImagePattern,
+				"launchType":   ""}}
```

Comment:
this caused an error in the e2e testing since it tried to set the `LaunchType` stack parameter to empty string...but that param doesn't exist in the ec2 provider template.  maybe update line `256` to only set the value iff launchType is in the map and not empty string?